### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by sohan9819**
Added authentication , form validation and private routes 

test user credentials 
username : adarshbalika
password : adarshbalika

### CODE REVIEW

**Review by rahul31agrawal — APPROVED** [review]
Hi Sohan,

Looks good to merge

File: src/routes/routes.jsx [hunk 1]
```diff
@@ -9,17 +9,40 @@ import {
...
-      <Route path='/home' element={<Home />} />
-      <Route path='/explore' element={<Explore />} />
-      <Route path='/bookmarks' element={<Bookmarks />} />
-      <Route path='/profile' element={<Profile />} />
-      <Route path='/auth-signin' element={<SignIn />} />
-      <Route path='/auth-signup' element={<SignUp />} />
+      <Route
+        path='/home'
+        element={authStatus ? <Home /> : <Navigate to={'/auth/signin'} />}
+      />
```
Comment (vishalg8454) [inline src/routes/routes.jsx:25]: You can convert into a `RequireAuth` component.

**Review by vishalg8454 — APPROVED** [review]
Good UI, can merge. One suggestion - add some test user login facility.

File: src/routes/routes.jsx (hunk 1 above)
Comment (sohan9819) [reply src/routes/routes.jsx:25]: thanks for the suggestions , I have added those in my next pr 👍

**Pull request merged**